### DISCUSSION TITLE
A current version refactor with significant breaking changes

### DISCUSSION THREAD

**Discussion by gdgellatly** [OCA/odoo-community.org#105] [Q&A]
Hi all,

I have a module I've maintained for many many years. It's current latest version is v14. In the process of migrating to v16, we discovered a different way via an OCA module to achieve the same outcome which is much nicer. Due to some other constraints, primarily _inherit and it's behaviour with many2many fields using relation keyword, I must use the same module name for v14.

It is so different that the diff will be 100%, usage entirely different, views. And the original module will be kept for v14 with the _original suffix.

I am just finishing rewriting it, just doing migration scripts, however do we have a way to significantly warn users who try to upgrade. I know within OCA modules we have had quite a few breaking changes and significant refactors introduced for modules with no warning, so it would be nice to hear peoples ideas on this.

**Comment by Shide** [comment]
If you think old users (v14 to v16) must be clarified, put the full instructions on the readme folder and that's it.

**Comment by theangryangel** [comment]
> It is so different that the diff will be 100%, usage entirely different, views. And the original module will be kept for v14 with the _original suffix.

I’ve come to the conclusion that no matter how much you document changes someone will always be surprised.

On that basis if it’s functionally the same, but there’s a risk of data loss and it’s difficult/impossible to create the relevant migrations, I’ve been pushing team members at work to create a new module, effectively leave the old one to die off and leave a big fat warning on the README for the module that it should be considered vestigial (fixes only), not to forward port it to future, and to look at the other module on-going. That way its conscious opt-in choice.

**Reply by francesco-ooops** [reply]
Agreed. From a functional POV, what helps the most when considering module migration is:

- stating in readme of old module that is will be discontinued in future versions
- stating in readme of new module that it replaces the old module
- in migration issue(s) for later version(s), stating that the previous module has been discontinued in favor of one or more other modules

**Comment by liebana** [comment]
+1 to new module, letting the other die.
